src/ar_kernel.cpp

```diff
@@ -591,23 +561,14 @@ uint32_t ar_get_system_load(void)
 // See ar_kernel.h for documentation of this function.
 uint32_t ar_get_tick_count(void)
 {
-#if AR_ENABLE_TICKLESS_IDLE
-    uint32_t elapsed_ticks = ar_port_get_timer_elapsed_us() / (kSchedulerQuanta_ms * 1000);
-    return g_ar.tickCount + elapsed_ticks;
-#else
-    return g_ar.tickCount;
-#endif // AR_ENABLE_TICKLESS_IDLE
+    return ar_port_get_time_absolute_ticks();
 }
 
 // See ar_kernel.h for documentation of this function.
 uint32_t ar_get_millisecond_count(void)
 {
-#if AR_ENABLE_TICKLESS_IDLE
-    uint32_t elapsed_ms = ar_port_get_timer_elapsed_us() / 1000;
-    return g_ar.tickCount * ar_get_milliseconds_per_tick() + elapsed_ms;
-#else
-    return g_ar.tickCount * ar_get_milliseconds_per_tick();
-#endif // AR_ENABLE_TICKLESS_IDLE
+    return ar_port_get_time_absolute_ms();
+    // return ar_port_get_time_absolute_ticks() * kSchedulerQuanta_ms; //do we have to limit resolution?
```

Comment:
`ar_get_tick_count()`  and `ar_get_millisecond_count()` should probably be inline, in `ar_internal.h`.

Reply:
Yeah, forgot to inline them. Even though, compiler would probably inline them anyway.
Well, I think that application should have access to system time. Also application should not have access to port functions.